contracts/orbital-core/src/contract.rs

```diff
@@ -85,9 +98,26 @@ pub fn query(deps: QueryDeps, env: Env, msg: QueryMsg) -> StdResult<Binary> {
             to_json_binary(&query_clearing_account(deps, domain, addr)?)
         }
         QueryMsg::Balance { query_id } => to_json_binary(&query_icq_balance(deps, env, query_id)?),
+        QueryMsg::IcqTransfersAmount {} => to_json_binary(&query_transfers_number(deps)?),
+        QueryMsg::IcqRecipientTxs { recipient } => {
+            to_json_binary(&query_recipient_txs(deps, recipient)?)
+        }
     }
 }
 
+fn query_recipient_txs(deps: QueryDeps, recipient: String) -> StdResult<RecipientTxsResponse> {
+    let txs = RECIPIENT_TXS
+        .may_load(deps.storage, recipient)?
+        .unwrap_or_default();
+
+    Ok(RecipientTxsResponse { transfers: txs })
+}
```

Comment:
in this pr, this is just a storage item that gets updated via the `sudo` endpoint. in practice we will likely want to keep pruning these state variables or have some key-value logic to avoid loading too much, perhaps something along the lines of a composite key with block height & recipient as the key.